src/new-components/heading/heading.tsx

```diff
@@ -21,15 +23,10 @@ type HeadingProps = SupportedHeadingElementProps & {
     lineClamp?: 1 | 2 | 3 | 4 | 5 | '1' | '2' | '3' | '4' | '5'
 }
 
-function Heading({
-    level,
-    weight = 'regular',
-    size,
-    tone = 'normal',
-    children,
-    lineClamp,
-    ...props
-}: HeadingProps) {
+const Heading = forwardRefWithAs<HeadingProps>(function Heading(
```

Comment:
We can and should do it with a regular `React.forwardRef` call, as this component does not support allowing the caller to pass in a `component` prop and change the element it renders with.

Reply:
Makes sense 👍